### ISSUE TITLE
a bug

### ISSUE THREAD

**Issue by 9001** [9001/r0c#2]
resize a netcat (10 lines > 4 lines > 5 lines or something) then pgup/pgdn
```
005352              __
005352    _      __/ /_  ____  ____  ____  _____
005352   | | /| / / __ \/ __ \/ __ \/ __ \/ ___/
005352   | |/ |/ / / / / /_/ / /_/ / /_/ (__  )
005352   |__/|__/_/ /_/\____/\____/ .___/____/
005352                           /_/
005352 Traceback (most recent call last):
005352   File "C:\Program Files\Python37\lib\asyncore.py", line 83, in read
005352     obj.handle_read_event()
005352   File "C:\Program Files\Python37\lib\asyncore.py", line 422, in handle_read_event
005352     self.handle_read()
005352   File "C:\Users\ed\dev\r0c\r0c\inetcat.py", line 107, in handle_read
005352     self.read_cb(False, len(src))
005352   File "C:\Users\ed\dev\r0c\r0c\ivt100.py", line 2595, in read_cb
005352     self.refresh(old_cursor != self.linepos)
005352   File "C:\Users\ed\dev\r0c\r0c\ivt100.py", line 738, in refresh
005352     to_send += self.update_chat_view(full_redraw, mark_messages_read)
005352   File "C:\Users\ed\dev\r0c\r0c\ivt100.py", line 1393, in update_chat_view
005352     ref, ref.cdr, len(ref.txt), 0, len(ref.txt) - ref.cdr, ch
005352   File "C:\Users\ed\dev\r0c\r0c\chat.py", line 158, in c_segm
005352     self.unformatted = self.txt[0]
005352 IndexError: list index out of range
```

**Comment by 9001** [comment]
should have been fixed by 109b65f82023240a423bdf97dbdd382a980b2336

**Comment by 9001** [comment]
(the scrolling stuff is still terrible and really needs a rewrite)
